Disables timezone options if it is set to automatic

priyeshshah11 opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->
This PR disables all the timezone options in the list if timezone is set to automatic. It also adds `componentDidUpdate` in order to update the timezone options when automatic is toggled on/off as navigating back doesn't unmount the page, thus it won't update the timezone options & stay disabled without this.

### Fixed Issues
<!---
1. Please replace GH_LINK with a URL link to the GitHub issue this Pull Request is fixing.
2. Please replace PROPOSAL: GH_LINK_ISSUE(COMMENT) with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<number-of-the-issue>
$ https://github.com/Expensify/App/issues/<number-of-the-issue(comment)>

Do NOT only link the issue number like this: $ #<number-of-the-issue>
--->
$ https://github.com/Expensify/App/issues/14340   
PROPOSAL: https://github.com/Expensify/App/issues/14340#issuecomment-1424471308


### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

1. Go to Settings -> Profile -> Timezone.
2. Toggle ON automatic & add `/select` to the existing route in the browser.
3. Now, try selecting items from the list - they should be disabled.
4. Try searching for items in the list with your country/city name & verify it filters the list accordingly but the options are still not selectable
5. Now go back.
6. Toggle OFF automatic & click on the selected timezone.
7. Try searching for items in the list with your country/city name & verify it filters the list accordingly and the options are now enabled.
8. Now, try selecting an item from the list & verify it gets selected & you're navigated back.
- [x] Verify that no errors appear in the JS console

### Offline tests
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->

1. Go to Settings -> Profile -> Timezone.
2. Toggle OFF automatic & click on the selected timezone.
3. Try searching for items in the list with your country/city name & verify it filters the list accordingly and the options are now enabled.
4. Now, try selecting an item from the list & verify it gets selected & you're navigated back.
- [x] Verify that no errors appear in the JS console

You won't be able to go to the timezone options page by modifying the URL in offline mode

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

Same as **Tests** section
- [x] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

<!-- add screenshots or videos here -->

https://user-images.githubusercontent.com/38547776/219102821-485d5a8e-e2bf-4b1e-92e0-47ae3f611fb5.mov

</details>

<details>
<summary>Mobile Web - Chrome</summary>

<!-- add screenshots or videos here -->

https://user-images.githubusercontent.com/38547776/219114458-3c995f5f-249d-4599-8a28-1cb4f38a7632.mov


https://user-images.githubusercontent.com/38547776/219114486-83d7d781-9206-4569-aca1-7653fb3f86b2.mov


</details>

<details>
<summary>Mobile Web - Safari</summary>

<!-- add screenshots or videos here -->
https://user-images.githubusercontent.com/38547776/219103013-63513cdb-7bb2-4809-b530-98e083f6dd6a.mov

</details>

<details>
<summary>Desktop</summary>

<!-- add screenshots or videos here -->

https://user-images.githubusercontent.com/38547776/219104857-b689b3bd-f850-4014-acf4-941b491efdff.mov


</details>

<details>
<summary>iOS</summary>

<!-- add screenshots or videos here -->

https://user-images.githubusercontent.com/38547776/219109151-616f55c8-9708-4e6b-9269-ea827b9d3cd5.mov


</details>

<details>
<summary>Android</summary>

<!-- add screenshots or videos here -->


https://user-images.githubusercontent.com/38547776/219114539-aed35175-a87a-413f-ba64-9d89f56c2355.mov


</details>

priyeshshah11 committed:
disabled timezone options if it is set to automatic

mollfpr commented:
There's a bug I found while testing this PR. Whenever I redirect to the select timezone page with _automatic_ set to `false`, select a timezone. After the timezone is updated, go back to the select timezone page, not select the current timezone.

https://user-images.githubusercontent.com/25520267/219404719-5f4baa2f-3751-4f33-bcb2-6299b5073d6a.mp4

Also, I confirm this already in staging, so no regression tied to this PR.

mollfpr commented:
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android / native
    - [ ] Android / Chrome
    - [ ] iOS / native
    - [ ] iOS / Safari
    - [ ] MacOS / Chrome / Safari
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Mobile Web - Chrome</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Mobile Web - Safari</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Desktop</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Android</summary>

<!-- add screenshots or videos here -->

</details>

mollfpr commented on src/pages/settings/Profile/TimezoneSelectPage.js:
```diff
@@ -34,27 +34,48 @@ class TimezoneSelectPage extends Component {
+            timezoneInputText: this.timezone.selected,
+            timezoneOptions: updatedAllTimezones,
+        });
+    }
```
@priyeshshah11  Why do we need this `componentDidUpdate`?

mollfpr replied:
>It also adds componentDidUpdate in order to update the timezone options when automatic is toggled on/off as navigating back doesn't unmount the page, thus it won't update the timezone options & stay disabled without this.

We should put that instead of _// Update timezoneInputText & all timezoneOptions when the timezone object changes_

priyeshshah11 replied:
@mollfpr 
>It also adds componentDidUpdate in order to update the timezone options when automatic is toggled on/off as navigating back doesn't unmount the page, thus it won't update the timezone options & stay disabled without this.

priyeshshah11 replied:
Done

mollfpr replied:
Thanks @priyeshshah11 that will help to explain what's going on 👍

priyeshshah11 committed:
updated the comment in componentDidUpdate

amyevans commented:
Assigning @tylerkaraszewski since a bug caused me to be added to https://github.com/Expensify/App/issues/14340 by mistake, but Tyler is CME on that GH, not me

mollfpr commented on src/pages/settings/Profile/TimezoneSelectPage.js:
```diff
@@ -34,27 +34,49 @@ class TimezoneSelectPage extends Component {
+        // componentDidUpdate is added in order to update the timezone options when automatic is toggled on/off as
+        // navigating back doesn't unmount the page, thus it won't update the timezone options & stay disabled without this.
+        const newTimezone = lodashGet(this.props.currentUserPersonalDetails, 'timezone', CONST.DEFAULT_TIME_ZONE);
+        if (_.isEqual(this.timezone, newTimezone)) { return; }
```
```suggestion
        if (_.isEqual(this.timezone, newTimezone)) {
            return;
        }

```

mollfpr commented:
@priyeshshah11 Is it possible to DRY the below codes? 
```
lodashGet(props.currentUserPersonalDetails, 'timezone', CONST.DEFAULT_TIME_ZONE);
```

```
timezone => ({
                text: timezone,
                keyForList: timezone,

                // Include the green checkmark icon to indicate the currently selected value
                customIcon: timezone === this.timezone.selected ? greenCheckmark : undefined,

                // This property will make the currently selected value have bold text
                boldStyle: timezone === this.timezone.selected,
            })
```
---

Example:
```js
    getUserTimezone(currentUserPersonalDetails) {
        return lodashGet(currentUserPersonalDetails, 'timezone', CONST.DEFAULT_TIME_ZONE);
    }

    formatTimezone(timezone) {
        return {
            text: timezone,

            keyForList: timezone,

            // Include the green checkmark icon to indicate the currently selected value
            customIcon: timezone === this.timezone.selected ? greenCheckmark : undefined,

            // This property will make the currently selected value have bold text
            boldStyle: timezone === this.timezone.selected,
        };
    }
```

```diff
-        this.timezone = lodashGet(props.currentUserPersonalDetails, 'timezone', CONST.DEFAULT_TIME_ZONE);
+        this.timezone = this.getUserTimezone(props.currentUserPersonalDetails);
         this.allTimezones = _.chain(moment.tz.names())
             .filter(timezone => !timezone.startsWith('Etc/GMT'))
-            .map(timezone => ({
-                text: timezone,
-                keyForList: timezone,
-
-                // Include the green checkmark icon to indicate the currently selected value
-                customIcon: timezone === this.timezone.selected ? greenCheckmark : undefined,
-
-                // This property will make the currently selected value have bold text
-                boldStyle: timezone === this.timezone.selected,
-            }))
+            .map(this.formatTimezone)
             .value();

         this.state = {
@@ -58,23 +49,36 @@ class TimezoneSelectPage extends Component {
     componentDidUpdate() {
         // componentDidUpdate is added in order to update the timezone options when automatic is toggled on/off as
         // navigating back doesn't unmount the page, thus it won't update the timezone options & stay disabled without this.
-        const newTimezone = lodashGet(this.props.currentUserPersonalDetails, 'timezone', CONST.DEFAULT_TIME_ZONE);
-        if (_.isEqual(this.timezone, newTimezone)) { return; }
+        const newTimezone = this.getUserTimezone(this.props.currentUserPersonalDetails);
+        if (_.isEqual(this.timezone, newTimezone)) {
+            return;
+        }
+
         this.timezone = newTimezone;
-        const updatedAllTimezones = _.map(this.allTimezones, timezone => ({
-            ...timezone,
+        const updatedAllTimezones = _.map(this.allTimezones, this.formatTimezone);
+
+        this.setState({
+            timezoneInputText: this.timezone.selected,
+            timezoneOptions: updatedAllTimezones,
+        });
+    }
```

priyeshshah11 committed:
refactored componentDidUpdate to fix options not udating issue

priyeshshah11 committed:
refactored get timezone code to follow DRY

priyeshshah11 commented:
@mollfpr Thanks for that suggestion. I have refactored the get timezone part but not the `formatTimezone` part as I realised that there was an issue in that code where the `timezone` in both `map` represented different type of data.

Thus, I have refactored it to use the correct type of data in each `map`. I have also added a new `getKey` function to update  the key which includes the timestamp in it as the options in the list weren't getting updated properly when going back & forth between the timezone pages. This fixes the bug that you found & mentioned [here](https://github.com/Expensify/App/pull/15176#issuecomment-1433239168).

Let me know your thoughts on those changes! Thanks 😄

priyeshshah11 commented:
<img width="761" alt="Screen Shot 2023-02-18 at 7 14 24 AM" src="https://user-images.githubusercontent.com/38547776/219784265-71007f10-7918-40f9-9822-dd17c60e1628.png">

Sorry, I didn't mean to remove the review request for @tylerkaraszewski but re-requesting review from @mollfpr did it automatically & now I can't request from Tyler.

mollfpr commented on src/pages/settings/Profile/TimezoneSelectPage.js:
```diff
@@ -34,27 +34,71 @@ class TimezoneSelectPage extends Component {
+     */
+    getKey(text) {
+        return `${text}-${(new Date()).getTime()}`;
+    }
```
Could you add a description of why we are doing this?

mollfpr commented on src/pages/settings/Profile/TimezoneSelectPage.js:
```diff
@@ -34,27 +34,71 @@ class TimezoneSelectPage extends Component {
+
+                // This property will make the currently selected value have bold text
+                boldStyle: text === this.timezone.selected,
+            };
```
We have a similar object structured within the constructor. We could probably create a function to return the object with a single argument.

mollfpr commented:
Thanks for the update @priyeshshah11! It solves the issue I encountered before, just left with a comment.

priyeshshah11 committed:
Code refactoring to follow DRY & add comments

priyeshshah11 commented:
> Thanks for the update @priyeshshah11! It solves the issue I encountered before, just left with a comment.

@mollfpr Done, refactored as requested 👍

mollfpr commented on src/pages/settings/Profile/TimezoneSelectPage.js:
```diff
@@ -33,28 +33,74 @@ class TimezoneSelectPage extends Component {
-                // This property will make the currently selected value have bold text
-                boldStyle: timezone === this.currentSelectedTimezone,
-            }))
+            .map(timezone => this.getTimezoneOption(timezone))
```
```suggestion
            .map(this.getTimezoneOption)
```

priyeshshah11 committed:
minor code cleanup

mollfpr reviewed (dismissed):
LGTM 👍

priyeshshah11 commented:
@mollfpr @tylerkaraszewski Are we waiting on anything here? Is there any action required from me?

mollfpr commented:
@priyeshshah11 we just need to wait for @tylerkaraszewski to review this.

mollfpr commented:
@priyeshshah11 Could you resolve the conflict? Thanks!

priyeshshah11 committed:
merged master & resolved conflicts

priyeshshah11 commented:
> @priyeshshah11 Could you resolve the conflict? Thanks!

Done @mollfpr

mollfpr commented:
Bump @tylerkaraszewski for final review.

mollfpr commented:
Bump @tylerkaraszewski for review.

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/4318706659) to staging by https://github.com/tylerkaraszewski in version: 1.2.78-0 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/4344937447) to production by https://github.com/mountiny in version: 1.2.78-0 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

sobitneupane commented:
[This regression](https://github.com/Expensify/App/issues/16905) was caused by this PR. `this.currentSelectedTimezone` was removed in this PR but was missed to remove in the following:

https://github.com/Expensify/App/blob/10d3d52e993303db477478d323af925f4d439acf/src/pages/settings/Profile/TimezoneSelectPage.js#L142
